codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -948,6 +948,12 @@ private String[] splitInitialLine(ByteBuf asciiBuffer) {
 
         final int end = startContent + asciiBuffer.readableBytes();
 
+        if (isControlOrWhitespaceAsciiChar(asciiBytes[end - 1])) {
+            // There should no extra control or whitespace char.
+            // See https://datatracker.ietf.org/doc/html/rfc2616#section-5.1
+            throw new IllegalArgumentException();
```

Comment:
ffti:
```suggestion
            throw new IllegalArgumentException("Illegal character in request line: 0x" + Integer.ToHexString(char));
```

Reply:
+1 for a hex representation and index of its position

Reply:
Good idea!